src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -303,13 +304,21 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 		 * see nodeAgg.c:build_pertrans_for_aggref(). In that case we use
 		 * sorted agg strategy, otherwise we use hash strategy.
 		 */
-		if (!groupingIsHashable || hasDistinctAggregate)
+		if (!enable_hashagg || !groupingIsHashable || hasDistinctAggregate)
 		{
+			char *messageHint = NULL;
+			if (!enable_hashagg && groupingIsHashable)
+			{
+				messageHint = "Setting enable_hashagg to on may help "
+							  "if grouped column list is hashable.";
```

Comment:
```groupingIsHashable``` asserts that grouping here is hashable. We do not need to second hint line. Perhaps something like
```
"Consider setting enable_hashagg to on."
```